internal/provider/database/resource.go

```diff
@@ -51,6 +52,18 @@ func (state *DatabaseResourceModel) CheckReady() error {
 		return fmt.Errorf("Database %s/%s/%s has an unexpected state: expected=%s, found=%s",
 			state.Organization, state.Project, state.Name, expectedState, *state.Status.State)
 	}
+	// Check that DBA password is up-to-date if password update is supported
+	resp, err := state.UpdateDbaPassword(ctx, client, nil)
+	if IsDbaPasswordUpdateUnsupportedError(resp, err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("DBA password for database %s/%s/%s has not been updated",
+			state.Organization, state.Project, state.Name)
```

Comment:
I like the logic, but it would be nice to include in the error details that the most likely cause of the failure is that there was an external change that set the database password to not match the old or new password and to resolve it, manually set the password to the new one.

Reply:
But this is the case where an error was not returned by the REST service. If it did, we would have returned the error the user.

If we're here, then the response would have been "202 Accepted", which indicates that password rotation is still in progress. Password rotation could be in progress indefinitely if the database is in Stopped state, or if there is some issue that is preventing the operator from doing reconciliation (but not an authentication failure, since that would cause the operator to blow away `target` key).